pandas/_libs/sparse.pyx

```diff
@@ -448,7 +448,7 @@ cdef class BlockIndex(SparseIndex):
         ylen = y.blengths
 
         # block may be split, but can't exceed original len / 2 + 1
-        max_len = int(min(self.length, y.length) / 2) + 1
```

Comment:
are there no cases where `int` rounds up?

Reply:
No, in order for ```int()``` to round up it needs to be done with ```math.ceil``` or with a "hack" like:

```python
number // divider + (number % divider > 0)
```

---

For example:
```python

number = 42.01
divider = 8

number / divider # 5.25125
number // divider # 5.0
int(number / divider) # 5
number // divider + (number % divider > 0) # 6.0
```